vignettes/extending-jmpost.Rmd

```diff
@@ -294,6 +294,164 @@ object@fun(
+the `SimLongitudinalGSF` class is implemented:
+
+```R
+# Declare the new class
+.SimLongitudinalGSF <- setClass(
+    "SimLongitudinalGSF",
+    contains = "SimLongitudinal",
+    slots = c(
+        sigma = "numeric",
+        mu_s = "numeric",
+        mu_g = "numeric",
+        mu_b = "numeric",
+        a_phi = "numeric",
+        b_phi = "numeric",
+        omega_b = "numeric",
+        omega_s = "numeric",
+        omega_g = "numeric",
+        link_dsld = "numeric",
+        link_ttg = "numeric",
+        link_identity = "numeric"
+    )
+)
+
+# Define constructor function with sensible default values
+SimLongitudinalGSF <- function(
+    times = c(-100, -50, 0, 50, 100, 150, 250, 350, 450, 550) / 365,
+    sigma = 0.01,
+    mu_s = c(0.6, 0.4),
+    mu_g = c(0.25, 0.35),
+    mu_b = 60,
+    a_phi = c(4, 6),
+    b_phi = c(4, 6),
+    omega_b = 0.2,
+    omega_s = 0.2,
+    omega_g = 0.2,
+    link_dsld = 0,
+    link_ttg = 0,
+    link_identity = 0
+) {
+    .SimLongitudinalGSF(
+        times = times,
+        sigma = sigma,
+        mu_s = mu_s,
+        mu_g = mu_g,
+        mu_b = mu_b,
+        a_phi = a_phi,
+        b_phi = b_phi,
+        omega_b = omega_b,
+        omega_s = omega_s,
+        omega_g = omega_g,
+        link_dsld = link_dsld,
+        link_ttg = link_ttg,
+        link_identity = link_identity
+    )
+}
+
+sampleSubjects.SimLongitudinalGSF <- function(object, subjects_df) {
+    res <- subjects_df |>
+        dplyr::mutate(study_idx = as.numeric(.data$study)) |>
+        dplyr::mutate(arm_idx = as.numeric(.data$arm)) |>
+        dplyr::mutate(psi_b = stats::rlnorm(dplyr::n(), log(object@mu_b[.data$study_idx]), object@omega_b)) |>
+        dplyr::mutate(psi_s = stats::rlnorm(dplyr::n(), log(object@mu_s[.data$arm_idx]), object@omega_s)) |>
+        dplyr::mutate(psi_g = stats::rlnorm(dplyr::n(), log(object@mu_g[.data$arm_idx]), object@omega_g)) |>
+        dplyr::mutate(psi_phi = stats::rbeta(dplyr::n(), object@a_phi[.data$arm_idx], object@b_phi[.data$arm_idx]))
+    res[, c("pt", "arm", "study", "psi_b", "psi_s", "psi_g", "psi_phi")]
+}
+
+sampleObservations.SimLongitudinalGSF <- function(object, times_df) {
+    times_df |>
+        dplyr::mutate(mu_sld = gsf_sld(.data$time, .data$psi_b, .data$psi_s, .data$psi_g, .data$psi_phi)) |>
+        dplyr::mutate(dsld = gsf_dsld(.data$time, .data$psi_b, .data$psi_s, .data$psi_g, .data$psi_phi)) |>
+        dplyr::mutate(ttg = gsf_ttg(.data$time, .data$psi_b, .data$psi_s, .data$psi_g, .data$psi_phi)) |>
+        dplyr::mutate(sld = stats::rnorm(dplyr::n(), .data$mu_sld, .data$mu_sld * object@sigma)) |>
+        dplyr::mutate(
+            log_haz_link =
+                (object@link_dsld * .data$dsld) +
+                (object@link_ttg * .data$ttg) +
+                (object@link_identity * .data$mu_sld)
+        )
+}
+```
+
+The `subjects_df` argument to the `sampleSubjects` method is a `data.frame` with the following columns:
+- `pt` - The subject identifier
```

Comment:
is this already consistent with the names elsewhere? Just because "pt" sounds more like "patient" instead of "subject".